pkg/handler/repository_test.go

```diff
@@ -9,17 +9,77 @@ import (
 	"testing"
 
 	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/dao"
 	"github.com/content-services/content-sources-backend/pkg/db"
 	"github.com/content-services/content-sources-backend/pkg/models"
 	"github.com/content-services/content-sources-backend/pkg/seeds"
 	"github.com/labstack/echo/v4"
 	"github.com/redhatinsights/platform-go-middlewares/identity"
 	"github.com/rs/zerolog/log"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/suite"
 	"gorm.io/gorm"
 )
 
+type MockRepositoryDao struct {
+	mock.Mock
+}
+
+func (r *MockRepositoryDao) Create(newRepo api.RepositoryRequest) error {
+	args := r.Called(newRepo)
+	return args.Error(0)
+}
+func (r *MockRepositoryDao) Fetch(orgId string, uuid string) api.RepositoryResponse {
+	args := r.Called(orgId, uuid)
+	if args.Get(0) == nil {
+		return api.RepositoryResponse{}
+	}
+	rr, ok := args.Get(0).(api.RepositoryResponse)
+	if ok {
+		return rr
+	} else {
+		return api.RepositoryResponse{}
+	}
+}
+
+func (r *MockRepositoryDao) Update(orgId string, uuid string, repoParams api.RepositoryRequest) error {
+	args := r.Called(orgId, uuid, repoParams)
+	return args.Error(0)
+}
```

Comment:
This is a 'mock' version of the dao layer, it lets us test the api without actually invoking the database.  So the goal is for the dao to have a very well defined interface, and we can test each layer separately.